### PR TITLE
Book Library Page FE

### LINKED ISSUE #25: Front-End Layouts
create css and js for :

* admin control panel
    * login
    * homepage
    * books
    * users
    * borrowing section
* website landing page   
* error pages

### LINKED ISSUE #35: Book Library Page
Relates #31 

### libraryBook.hbs: #26 
 - Page contain table view the data
    * name book
    * name author
    * serial number
    * book status
    * description this is button view 
        *  button view --> view the all details for single book
 - Page contain input search.
 

### libraryBook.js
- File contain one functions get #10 

 * Get :
    * Render libraryBook.Page  .
 * **response**:
      * data in DB #16
          * SELECT
             getLibraryBooks () --> return array of object all data of library 
```js
{
statusCode: 200,
response : [ {  } ];
}
```

### PR DESCRIPTION
**Pull request by asmaa-ezz**
## Description

add a the search function that search for the book name or book author 

### Type of change

New features #34 
Description :
Add function search in front-end admin.js 

Fixes #35 
Description : 
Modify view_books.hbs to convert static view to dynamic for library books

### CODE REVIEW

**Review by ali-7 — APPROVED** [review]

File: src/views/layouts/admin.hbs [hunk 1]
```diff
@@ -11,9 +11,12 @@
     {{>admin_sidebar}}
     {{{body}}}
   </main>
+  {{#if admin}}
```
Comment (ali-7) [inline src/views/layouts/admin.hbs]: but admin.js without if, coz we need it in all the project

**Review by ali-7 — CHANGES_REQUESTED** [review]

File: src/views/layouts/admin.hbs (hunk 1 above)
Comment (asmaa-ezz) [reply src/views/layouts/admin.hbs]: solve

**Review by RamyAlshurafa — DISMISSED** [review]
For next time:
## use descriptive commits

File: public/js/admin.js [hunk 2]
```diff
@@ -32,4 +32,17 @@ const search = (input, table1) => {
...
+};
+
+const check = (input, errorMessageElement, errMessage) => {
+  if (!input.value) {
+    displayErr(errorMessageElement, errMessage);
+  } else {
+    displayErr(errorMessageElement, '');
+    return true;
+  }
+>>>>>>> 0140ecb70bb74a3a3134c3e58d5570d9de0420b1
```
Comment (RamyAlshurafa) [inline public/js/admin.js]: ### Did you solve the conflict?

File: public/js/book.js [hunk 3]
```diff
@@ -7,6 +7,22 @@ const library = document.getElementById('table-library');
...
+const addBook = document.getElementById('addBook');
+const categoryError = document.getElementById('categoryError');
+const nameBookError = document.getElementById('nameBookError');
+const nameAuthorError = document.getElementById('nameAuthorError');
+const nameCategory = document.getElementById('nameCategory');
+const serialNumber = document.getElementById('serialNumber');
+const addCategory = document.getElementById('addCategory');
+const serialNumberCategoryError = document.getElementById('serialNumberCategoryError');
+const nameCategoryError = document.getElementById('nameCategoryError');
+>>>>>>> 0140ecb70bb74a3a3134c3e58d5570d9de0420b1
```
Comment (RamyAlshurafa) [inline public/js/book.js]: ### conflict

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]
### Conflict in the PR
check all files

**Review by ali-7 — APPROVED** [review]

**Review by lubnaabd — APPROVED** [review]

File: public/js/book.js [hunk 4]
```diff
@@ -2,6 +2,11 @@ const mainBook = document.getElementsByClassName('main__conten--book');
 const tabLinks = document.getElementsByClassName('main__header--nav-tablink');
 const next = document.getElementById('next');
 const AddCategory = document.getElementById('button-add-category');
+const searchLibrary = document.getElementById('input-search-library');
+const library = document.getElementById('table-library');
+
+search(searchLibrary, library);
+
 const nameBook = document.getElementById('nameBook');
```
Comment (RamyAlshurafa) [inline public/js/book.js:7]: declare all your variables at the top of your file

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]
Declare all variables at the top of the file

**Review by RamyAlshurafa — APPROVED** [review]
Great Job :+1:

**Pull request merged**